adadamp/_dist.py

```diff
@@ -311,3 +312,46 @@ def _get_gradients(
         grads = [grad_fn(idx=idx) for idx in worker_idxs]
         out = dask.compute(*grads)
         return out
+
+class DaskBaseDamper2(DaskBaseDamper):
+    
+    def fit(self, data, target):
+        start = time.time()
+        super().fit(data, target)
+        
+        stat = {
+            "n_data": self.meta_['n_data'],
+            "n_updates": self.meta_['n_updates'],
+            "epoch_time": time.time() - start,
+            "batch_size": self.batch_size_()
+        }
+        
+        if not hasattr(self, 'stats_'):
+            self.stats_ = []
+        self.stats_.append(stat)
+        
+    def get_stats(self):
+        return self.stats_
+    
+    
+    def score(self, X, y):
```

Comment:
This is finally working! However, there is a massive performance issue with it for some reason. When the code on line 347 and bellow is gone, it runs normally, but if it is there, the second epoch takes forever to run. 

I had a loop within the top level loop before, but i spent time vectorizing it and it is still slow. Any ideas why?

Reply:
spent a little more time looking at this today and still cant figure out why that code is going so slow. Do you have any tips for performance debugging pytorch stuff like this (or even performance debugging in general)?

Reply:
My favorite method is to throw the main function call into Dask. The Dask dashboard has a profiling pane, which shows which functions are taking the longest. This is the dashboard pane I look at: https://distributed.dask.org/en/latest/diagnosing-performance.html#statistical-profiling

Before:
``` python
def main():
    ...

if __name__ == "__main__":
    main()
```

With debugging:

``` python
from dask.distributed import Client
def main():
    ...

if __name__ == "__main__":
    client = Client()
    result = client.gather(client.submit(main()))
    # in browser, localhost:8787/profile to see timing
```

Reply:
Note:
- Add a no_grad to the score 2 method
- Dont loop over training set twice, copy code from score 1 and dont call super